### PR TITLE
feat: add new plugin type for cache manager

### PR DESCRIPTION
**Pull request by guillaumelamirand**
**Issue**

https://gravitee.atlassian.net/browse/ARCHI-231

**Description**

Following the work done on cluster, now make cache manager as plugin.

---
**Gravitee.io Automatic Deployment**

🚀 A prerelease version of this package has been published on Gravitee's private artifactory, you can:
 - use it directly by updating your project with version: `3.1.0-archi-231-cache-manager-SNAPSHOT`
 - download it from Artifactory [here](https://odbxikk7vo-artifactory.services.clever-cloud.com/gravitee-snapshots/io/gravitee/node/gravitee-node/3.1.0-archi-231-cache-manager-SNAPSHOT/gravitee-node-3.1.0-archi-231-cache-manager-SNAPSHOT.zip)

### CODE REVIEW

**Review by remibaptistegio — CHANGES_REQUESTED** [review]

File: gravitee-node-cache/gravitee-node-cache-common/src/main/java/io/gravitee/node/plugin/cache/common/InMemoryCache.java [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+                            } else {
+                                listener.onEntryEvicted(typeKey, value);
+                            }
+                            break;
+                        case EXPIRED:
+                            listener.onEntryExpired(typeKey, value);
+                            break;
+                        case REPLACED:
+                        case COLLECTED:
+                        case SIZE:
```
[the remark this answers is not in the document]
Comment (guillaumelamirand) [reply gravitee-node-cache/gravitee-node-cache-common/src/main/java/io/gravitee/node/plugin/cache/common/InMemoryCache.java:82]: I can move this and handle it the same way I do with expired, wdyt?

File: gravitee-node-cache/gravitee-node-cache-plugin-standalone/pom.xml [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        <groupId>io.gravitee.node</groupId>
+        <artifactId>gravitee-node-cache</artifactId>
+        <version>3.1.0-archi-231-cache-manager-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>gravitee-node-cache-plugin-standalone</artifactId>
+    <name>Gravitee.io - Node - Cache - Plugin - Standalone</name>
+    <description>Standalone Cache is the default plugin used for in-memory caching.</description>
+    <properties>
+        <caffeine.version>2.5.5</caffeine.version>
```
[the remark this answers is not in the document]
Comment (guillaumelamirand) [reply gravitee-node-cache/gravitee-node-cache-plugin-standalone/pom.xml:34]: no more yes!

File: gravitee-node-cache/gravitee-node-cache-common/src/main/java/io/gravitee/node/plugin/cache/common/InMemoryCache.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (guillaumelamirand) [reply gravitee-node-cache/gravitee-node-cache-common/src/main/java/io/gravitee/node/plugin/cache/common/InMemoryCache.java:82]: Ok I just use  onEntryEvicted

**Review by remibaptistegio — APPROVED** [review]

**Pull request merged**